build.go

```diff
@@ -0,0 +1,192 @@
+	"github.com/paketo-buildpacks/packit/v2/chronos"
+	"github.com/paketo-buildpacks/packit/v2/postal"
+	"github.com/paketo-buildpacks/packit/v2/sbom"
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+)
+
+//go:generate faux --interface EntryResolver --output fakes/entry_resolver.go
+type EntryResolver interface {
+	Resolve(string, []packit.BuildpackPlanEntry, []interface{}) (packit.BuildpackPlanEntry, []packit.BuildpackPlanEntry)
+	MergeLayerTypes(string, []packit.BuildpackPlanEntry) (launch, build bool)
+}
+
+//go:generate faux --interface DependencyManager --output fakes/dependency_manager.go
+type DependencyManager interface {
+	Deliver(dependency postal.Dependency, cnbPath, layerPath, platformPath string) error
+	GenerateBillOfMaterials(dependencies ...postal.Dependency) []packit.BOMEntry
+}
+
+//go:generate faux --interface VersionResolver --output fakes/version_resolver.go
+type VersionResolver interface {
+	Resolve(path string, entry packit.BuildpackPlanEntry, stack string) (postal.Dependency, error)
+}
+
+//go:generate faux --interface SBOMGenerator --output fakes/sbom_generator.go
+type SBOMGenerator interface {
+	GenerateFromDependency(dependency postal.Dependency, dir string) (sbom.SBOM, error)
+}
+
+//go:generate faux --interface ConfigParser --output fakes/config_parser.go
+type ConfigParser interface {
+	Parse(runtimeConfigFileGlob string) (string, error)
+}
+
+func Build(
+	entries EntryResolver,
+	dependencies DependencyManager,
+	versionResolver VersionResolver,
+	sbomGenerator SBOMGenerator,
+	configParser ConfigParser,
+	logger scribe.Emitter,
+	clock chronos.Clock,
+) packit.BuildFunc {
+	return func(context packit.BuildContext) (packit.BuildResult, error) {
+		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
+
+		// This no-op path is to accommodate for it the dotnet-publish buildpack
+		// generates a self-contained app
+		frameworkVersion, err := configParser.Parse(filepath.Join(context.WorkingDir, "*.runtimeconfig.json"))
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				logger.Process("Skipping build process as there was no runtimeconfig.json found")
+				logger.Break()
+				return packit.BuildResult{}, nil
+			}
+			return packit.BuildResult{}, err
+		}
+
+		if frameworkVersion == "" {
+			logger.Process("Skipping build process as there were no frameworks specified in the runtimeconfig.json")
+			logger.Break()
+			return packit.BuildResult{}, nil
+		} else {
+			context.Plan.Entries = append([]packit.BuildpackPlanEntry{
+				{
+					Name: "dotnet-core-aspnet-runtime",
+					Metadata: map[string]interface{}{
+						"version":        frameworkVersion,
+						"version-source": "runtimeconfig.json",
+					},
+				},
+			}, context.Plan.Entries...)
+		}
+
+		logger.Process("Resolving .NET Core ASP.NET Runtime version")
+
+		priorities := []interface{}{
+			"BP_DOTNET_FRAMEWORK_VERSION",
+			regexp.MustCompile(`.*\.(cs)|(fs)|(vb)proj`),
+			"runtimeconfig.json",
```

Comment:
```suggestion
			"BP_DOTNET_FRAMEWORK_VERSION",
			"runtimeconfig.json",
			regexp.MustCompile(`.*\.(cs)|(fs)|(vb)proj`),
```
I think the version extracted from the `runtimeconfig.json` should take precedence over the one from the projfile. Since the version in the `runtimeconfig.json` is our best and most up-to-date guess about what runtime version the app needs.